examples/apps/ai-collab/src/components/UserPresenceGroup.tsx

```diff
@@ -69,23 +58,35 @@ const UserPresenceGroup: React.FC<UserPresenceProps> = ({
 				// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 				const photoBlob = await client.api("/me/photo/$value").get();
 				// eslint-disable-next-line @typescript-eslint/no-unsafe-argument
-				const photoUrl = URL.createObjectURL(photoBlob);
-				setPhotoUrls((prevPhotos) => {
-					return [...prevPhotos, photoUrl];
-				});
-				userPresenceGroup.onlineUsers.local.set(`id-${uuid()}`, {
-					value: { photo: photoUrl },
-				});
-				setIsPhotoFetched(true);
+				photoUrl = URL.createObjectURL(photoBlob);
 			} catch (error) {
 				console.error(error);
 			}
-		};
+		}
+		// console.log("USER PRESENCES SHOULD ONLY BE SET ONCE");
+		userPresenceGroup.props.onlineUsers.local.set(`id-${uuid()}`, {
+			value: { photo: photoUrl },
+		});
 
-		if (!isPhotoFetched) {
-			fetchPhoto().catch((error) => console.error(error));
+		isFirstRender.current = false;
+	};
+
+	useEffect((): void => {
+		if (isFirstRender.current) {
+			updateUserPresenceGroup().catch((error) => console.error(error));
 		}
-	}, [isPhotoFetched, setIsPhotoFetched]);
+
+		const handleUpdate = (): void => {
+			// console.log(
+			// 	"On event updated??? Now it has items: ",
+			// 	userPresenceGroup.props.onlineUsers,
+			// );
+			setPhotoUrls(fetchAllPhotos());
+		};
+
+		userPresenceGroup.props.onlineUsers.events.on("updated", handleUpdate);
+		// handleUpdate();
```

Comment:
Calling `handleUpdate()` once here made sense to me, particularly for case where there are no changes (no "updated" event) in the presence group for a while... do we not need it?

Reply:
Based on testing, seems like we only need to update local photoUrls state, when presence has updated events.